native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceCallback.java

```diff
@@ -71,8 +72,10 @@ public void notifySuccess(Object result) {
             sendTextMessage((BString) result, promiseCombiner);
         } else if (result instanceof BArray) {
             sendBinaryMessage((BArray) result, promiseCombiner);
-        } else {
+        } else if (result == null) {
             webSocketConnection.readNextFrame();
+        } else if (resource.equals(WebSocketConstants.RESOURCE_NAME_ON_TEXT_MESSAGE)) {
+            sendTextMessage(StringUtils.fromString(result.toString()), promiseCombiner);
         }
```

Comment:
Better to have a else part as well with a debug/error log or something which could help when resolving issue to. understand the flow